[BUG] test_delta_multi_part_write_round_trip_unmanaged and test_delta_part_write_round_trip_unmanaged failed DATA_SEED=1701608331

**Describe the bug**
rapids_integration-scala213-pre_release-github, build ID: 16

spark340 failed `AssertionError: Different line counts in 00000000000000000000.json`
```
src.main.python.delta_lake_write_test.test_delta_multi_part_write_round_trip_unmanaged[Double][DATAGEN_SEED=1701608331, INJECT_OOM, IGNORE_ORDER, ALLOW_NON_GPU(DeserializeToObjectExec,ShuffleExchangeExec,FileSourceScanExec,FilterExec,MapPartitionsExec,MapElementsExec,ObjectHashAggregateExec,ProjectExec,SerializeFromObjectExec,SortExec)]
src.main.python.delta_lake_write_test.test_delta_part_write_round_trip_unmanaged[Double][DATAGEN_SEED=1701608331, INJECT_OOM, IGNORE_ORDER, ALLOW_NON_GPU(DeserializeToObjectExec,ShuffleExchangeExec,FileSourceScanExec,FilterExec,MapPartitionsExec,MapElementsExec,ObjectHashAggregateExec,ProjectExec,SerializeFromObjectExec,SortExec)]
```


**Steps/Code to reproduce bug**
Please provide a list of steps or a code sample to reproduce the issue.
Avoid posting private or sensitive data.

**Expected behavior**
A clear and concise description of what you expected to happen.

**Environment details (please complete the following information)**
 - Environment location: [Standalone, YARN, Kubernetes, Cloud(specify cloud provider)]
 - Spark configuration settings related to the issue

**Additional context**
Add any other context about the problem here.

--- a/integration_tests/src/main/python/delta_lake_write_test.py
+++ b/integration_tests/src/main/python/delta_lake_write_test.py
@@ -32,9 +32,6 @@ delta_part_write_gens = [
     short_gen,
     int_gen,
     long_gen,
-    # Avoid NaNs since it falsely triggers switch to new file when checking if partition changed
-    FloatGen(no_nans=True),
-    DoubleGen(no_nans=True),
     # Some file systems have issues with UTF8 strings so to help the test pass even there
     StringGen('(\\w| ){0,50}'),
     boolean_gen,

